service/security/shiro/src/main/java/org/eclipse/kapua/service/authorization/role/shiro/RolePermissionServiceImpl.java

```diff
@@ -128,7 +128,7 @@ public void delete(KapuaId scopeId, KapuaId rolePermissionId) throws KapuaExcept
         KapuaLocator locator = KapuaLocator.getInstance();
         AuthorizationService authorizationService = locator.getService(AuthorizationService.class);
         PermissionFactory permissionFactory = locator.getFactory(PermissionFactory.class);
-        authorizationService.checkPermission(permissionFactory.newPermission(AuthorizationDomains.ROLE_DOMAIN, Actions.delete, scopeId));
+        authorizationService.checkPermission(permissionFactory.newPermission(AuthorizationDomains.ACCESS_INFO_DOMAIN, Actions.delete, scopeId));
```

Comment:
Why change this permission check? 

Since we this service manages the Permission assigned to the Role. So `role` was the correct permission to check. While `access_info` is the domain for the services/resources that are managing the assignment of the roles and permission to the user

Reply:
@Coduz Before this change the user had to have `ROLE_DOMAIN` delete permission for deleting a permission (not a role, but permission) on Roles tab. It just didn't seem right. So, after serious thought I changed it to `ACCESS_INFO_DOMAIN` delete permission. Was I wrong?

Reply:
Actually that was right, since those are the permission associated to the Role. 

The fact that those changes affetcs user's permission is a side effect. But the actual service to manage permission and roles associated to a user are the `AccessPermissionService` and the `AccessRoleService` that both require `access_info`.

As I said before

> [...] `access_info` is the domain for the services/resources that are managing the assignment of the roles and permission to the user. [...]

Do you agree?

Reply:
@Coduz Then this check should be changed to the previous state? The `role:delete` instead of the `access_info:delete` ? Than the affected permission delete button's enabled state should be changed as well, to need the `role:delete` permission.  Should the button need only `role:delete`? Or `access_info:delete` as well?
I wanted to confirm this with you, before making changes :)

Reply:
The only permission required to delete a Permission from a Role is `role:delete`.

So you can proceed to revert the change on the permission required in `RolePermissionServiceImpl` and change the permission checked when setting the enablement status of the Delete Button. You may also want to check that all permission on the Role view refers to the `RoleSessionPermission` except the enablement of the "Granted User" Tab, which requires `access_info:read`